libproxy/libproxyd.cpp

```diff
@@ -0,0 +1,177 @@
+/*******************************************************************************
+ * libproxy - A library for proxy configuration
+ * Copyright (C) 2015 Dominique Leuenberger <dimstar@opensuse.org>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301 USA
+ ******************************************************************************/
+
+#include <gio/gio.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+/* Import libproxy API */
+#include <proxy.h>
+
+/* ---------------------------------------------------------------------------------------------------- */
+
+static GDBusNodeInfo *introspection_data = NULL;
+
+/* Introspection data for the service we are exporting */
+static const gchar introspection_xml[] =
+  "<node>"
+  "  <interface name='org.libproxy.proxy'>"
```

Comment:
Hmm you know, you should really install this interface definition, and also a D-Bus service file to activate. Otherwise it's going to be real hard to debug with d-feet, etc. And the daemon would have to be already running somehow to work.

Alternatively, you could use your own D-Bus socket instead of using the session bus, but the benefit of auto-activating libproxyd seems worth it.

Reply:
oh, indeed; the daemon should be auto-started by a dbus service file. I knew I had some stuff missing :)